api.planx.uk/saveAndReturn/resumeApplication.test.ts

```diff
@@ -7,6 +8,10 @@ import { buildContentFromSessions } from "./resumeApplication";
 const ENDPOINT = "/resume-application";
 const TEST_EMAIL = "simulate-delivered@notifications.service.gov.uk"
 
+type DeepPartial<T> = T extends object ? {
+  [P in keyof T]?: DeepPartial<T[P]>;
+} : T;
```

Comment:
I'm going to do a bit of homework on it, but [type-fest](https://www.npmjs.com/package/type-fest) could prove to be a very handy package to integrate!